pkg/globalmanager/incrementallearningjob.go

```diff
@@ -0,0 +1,794 @@
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("Couldn't get key for object %+v: %v", obj, err))
+		return
+	}
+
+	backoff := time.Duration(0)
+	if !immediate {
+		backoff = getBackoff(jc.queue, key)
+	}
+
+	jc.queue.AddAfter(key, backoff)
+}
+
+// worker runs a worker thread that just dequeues items, processes them, and marks them done.
+// It enforces that the syncHandler is never invoked concurrently with the same key.
+func (jc *IncrementalJobController) worker() {
+	for jc.processNextWorkItem() {
+	}
+}
+
+func (jc *IncrementalJobController) processNextWorkItem() bool {
+	key, quit := jc.queue.Get()
+	if quit {
+		return false
+	}
+	defer jc.queue.Done(key)
+
+	forget, err := jc.sync(key.(string))
+	if err == nil {
+		if forget {
+			jc.queue.Forget(key)
+		}
+		return true
+	}
+
+	utilruntime.HandleError(fmt.Errorf("Error syncing incrementallearning job: %v", err))
+	jc.queue.AddRateLimited(key)
+
+	return true
+}
+
+// sync will sync the incrementallearning job with the given key if it has had its expectations fulfilled, meaning
+// it did not expect to see any more of its pods created or deleted. This function is not meant to be invoked
+// concurrently with the same key.
+func (jc *IncrementalJobController) sync(key string) (bool, error) {
+	startTime := time.Now()
+	defer func() {
+		klog.V(4).Infof("Finished syncing incrementallearning job %q (%v)", key, time.Since(startTime))
+	}()
+
+	ns, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return false, err
+	}
+	if len(ns) == 0 || len(name) == 0 {
+		return false, fmt.Errorf("invalid incrementallearning job key %q: either namespace or name is missing", key)
+	}
+	sharedIncrementalJob, err := jc.jobLister.IncrementalLearningJobs(ns).Get(name)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			klog.V(4).Infof("incrementallearning job has been deleted: %v", key)
+			return true, nil
+		}
+		return false, err
+	}
+	incrementaljob := *sharedIncrementalJob
+	// set kind for incrementaljob in case that the kind is None
+	incrementaljob.SetGroupVersionKind(neptunev1.SchemeGroupVersion.WithKind("IncrementalLearningJob"))
+	// incrementaljob first start, create pod for inference
+	if incrementaljob.Status.StartTime == nil {
+		now := metav1.Now()
+		incrementaljob.Status.StartTime = &now
+		pod := jc.getSpecifiedPods(&incrementaljob, "inference")
+		if pod == nil {
+			err = jc.createInferPod(&incrementaljob)
+		} else {
+			if pod.Status.Phase != v1.PodRunning && pod.Status.Phase != v1.PodPending {
+				err = jc.createInferPod(&incrementaljob)
+			}
+		}
+		if err != nil {
+			return false, nil
+		}
+	}
+
+	// if incrementaljob was finished previously, we don't want to redo the termination
+	if IsIncrementalJobFinished(&incrementaljob) {
+		return true, nil
+	}
+
+	forget := false
+	jobFailed := false
+	needUpdated := false
+
+	// update conditions of incremental job
+	needUpdated, err = jc.updateIncrementalJobConditions(&incrementaljob)
```

Comment:
pls log for err not used. but look like updateIncrementalJobConditions always returns nil.

Reply:
ok,i'll add log for it